docs/build_system.md

```diff
@@ -0,0 +1,70 @@
+# Build System
+
+There are many ways to build YARP, which means the build system is a bit more complicated than usual.
+
+## General Design
+
+1. Templates are generated by `templates/template.rb`
+2. `autoconf` create `./configure` and `autoheader` creates `config.h.in` (both files are platform-independent)
+3. `./configure` creates `include/yarp/config.h` (which contains `HAVE_*` macros, platform-specific) and the `Makefile`
```

Comment:
Removing autotools would remove these 2 layers, so that's the easiest change to simplify the build system.
Then the only generated files are the templates, that's it, and there is no need to know about autotools and which autotools-related files are platform-independent or plaftform-specific.